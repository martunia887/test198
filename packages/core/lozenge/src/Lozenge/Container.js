// @flow
/** @jsx jsx */
import { type Node } from 'react';
import { jsx } from '@emotion/core';
import { borderRadius } from '@atlaskit/theme';
import { type ThemeTokens } from '../theme';

const BORDER_RADIUS = `${borderRadius()}px`;

type ThemeTokensWithChildren = ThemeTokens & {
  children?: Node,
};

export default ({
  backgroundColor,
  textColor,
  children,
}: ThemeTokensWithChildren) => (
  <span
    css={{
      backgroundColor,
      borderRadius: BORDER_RADIUS,
      boxSizing: 'border-box',
      color: textColor,
      display: 'inline-block',
      fontSize: '11px',
      fontWeight: 700,
      lineHeight: 1,
      maxWidth: '100%',
      padding: '2px 0 3px 0',
      textTransform: 'uppercase',
      verticalAlign: 'baseline',
    }}
  >
    {children}
  </span>
);
