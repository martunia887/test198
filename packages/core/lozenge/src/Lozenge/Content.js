// @flow
/** @jsx jsx */
import { type Node } from 'react';
import { jsx } from '@emotion/core';
import { gridSize } from '@atlaskit/theme';
import { type ThemeTokens } from '../theme';

const HORIZONTAL_SPACING = `${gridSize() / 2}px`;

type ThemeTokensWithChildren = ThemeTokens & {
  children?: Node,
};

export default ({ maxWidth, children }: ThemeTokensWithChildren) => (
  <span
    css={{
      display: 'inline-block',
      verticalAlign: 'top',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      whiteSpace: 'nowrap',
      boxSizing: 'border-box',
      padding: `0 ${HORIZONTAL_SPACING}`,
      maxWidth: typeof maxWidth === 'number' ? `${maxWidth}px` : maxWidth,
      width: '100%',
    }}
  >
    {children}
  </span>
);
