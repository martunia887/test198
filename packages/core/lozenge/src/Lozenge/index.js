// @flow

import { type ThemeProp } from '@atlaskit/theme';
import React, { PureComponent, type Node } from 'react';
import Container from './Container';
import Content from './Content';
import {
  Theme,
  type ThemeAppearance,
  type ThemeProps,
  type ThemeTokens,
} from '../theme';

type Props = {
  /** The appearance type. */
  appearance: ThemeAppearance,

  /** Elements to be rendered inside the lozenge. This should ideally be just a word or two. */
  children?: Node,

  /** Determines whether to apply the bold style or not. */
  isBold: boolean,

  /** max-width of lozenge container. Default to 200px. */
  maxWidth: number | string,

  /** The theme the component should use. */
  theme?: ThemeProp<ThemeTokens, ThemeProps>,
};

export default class Lozenge extends PureComponent<Props> {
  static defaultProps = {
    isBold: false,
    appearance: 'default',
    maxWidth: 200,
  };

  render() {
    const { theme, children } = this.props;
    return (
      <Theme.Provider value={theme}>
        <Theme.Consumer {...this.props}>
          {tokens => (
            <Container {...tokens}>
              <Content {...tokens}>{children}</Content>
            </Container>
          )}
        </Theme.Consumer>
      </Theme.Provider>
    );
  }
}
