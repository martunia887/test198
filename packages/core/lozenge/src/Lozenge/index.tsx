import * as ThemeProp from '@atlaskit/theme/ThemeProp';
import React, { PureComponent, ReactNode } from 'react';
import Container from './Container';
import Content from './Content';
import { Theme, ThemeAppearance, ThemeProps, ThemeTokens } from '../theme';

interface Props {
  /** The appearance type. */
  appearance: ThemeAppearance;

  /** Elements to be rendered inside the lozenge. This should ideally be just a word or two. */
  children?: ReactNode;

  /** Determines whether to apply the bold style or not. */
  isBold: boolean;

  /** max-width of lozenge container. Default to 200px. */
  maxWidth: number | string;

  /** The theme the component should use. */
  theme?: ThemeProp<ThemeTokens, ThemeProps>;
}

export default class Lozenge extends PureComponent<Props> {
  static defaultProps = {
    isBold: false,
    appearance: 'default',
    maxWidth: 200,
  };

  render() {
    const { theme, children } = this.props;
    return (
      <Theme.Provider value={theme}>
        <Theme.Consumer {...this.props}>
          {tokens => (
            <Container {...tokens}>
              <Content {...tokens}>{children}</Content>
            </Container>
          )}
        </Theme.Consumer>
      </Theme.Provider>
    );
  }
}
