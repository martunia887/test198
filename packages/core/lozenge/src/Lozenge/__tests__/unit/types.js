// @flow
import Lozenge from '../../..';

describe('Lozenge', () => {
  it('should do something', () => {
    const generatedSnapShot = `
      declare module '@atlaskit/lozenge' {
  
      }
    `;
    expect(generatedSnapShot).toMatchSnapshot();
  });
});
