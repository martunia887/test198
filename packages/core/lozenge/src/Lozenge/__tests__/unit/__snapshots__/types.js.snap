// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
"
    declare module '@atlaskit/lozenge' {

    }
  "
`;

exports[`Lozenge should do something 1`] = `
"
      declare module '@atlaskit/lozenge' {
  
      }
    "
`;
